File: src/main/java/org/alham/alhamfirst/mapper/UserMapper.java
package org.alham.alhamfirst.mapper;

import org.alham.alhamfirst.common.error.AlhamCustomErrorLog;
import org.alham.alhamfirst.dto.user.UserDTO;
import org.alham.alhamfirst.entity.User;
import org.alham.alhamfirst.util.AESUtil;
import org.springframework.stereotype.Component;

@Component
public class UserMapper {

    public UserDTO createUserDTOFromEntity(User user) throws Exception {
        return UserDTO.builder()
                .id(AESUtil.encrypt(user.getId().toString()))
                .name(user.getName())
                .age(user.getAge())
                .userType(user.getUserType())
                .email(user.getEmail())
                .build();
    }


    public User createUserFromDTO(UserDTO userDTO) throws Exception {
        return User.builder()
                .id(Long.parseLong(AESUtil.decrypt(userDTO.getId())))
                .name(userDTO.getName())
                .age(userDTO.getAge())
                .userType(userDTO.getUserType())
                .email(userDTO.getEmail())
                .build();
    }




}

File: src/main/java/org/alham/alhamfirst/mapper/UserMapper.kt
package org.alham.alhamfirst.mapper;

import org.alham.alhamfirst.domain.dto.user.UserDTO
import org.alham.alhamfirst.domain.entity.User
import org.alham.alhamfirst.util.AESUtil

class UserMapper {

    fun createUserDTOFromEntity(user : User) : UserDTO {
        return UserDTO(
                id= AESUtil.encrypt(user.id.toString()),
                uuid = user.uuid,
                oauthProvider = user.oauthProvider,
                oauthId = user.oauthId,
                name = user.name,
                age = user.age,
                userType = user.userType,
                email = user.email
        )
    }


    fun createUserFromDTO(userDTO : UserDTO) : User {
        return User(
                id = userDTO.getIdDecrypt(),
                uuid = userDTO.uuid,
                oauthProvider = userDTO.oauthProvider,
                oauthId = userDTO.oauthId,
                name = userDTO.name,
                age = userDTO.age,
                userType = userDTO.userType,
                email = userDTO.email
        )
    }

    fun addUserStatData(userDTO: UserDTO, statData: MutableMap<String, Double>) : UserDTO {
        userDTO.statData = statData
        return userDTO
    }



}
